spec/reservation_hub_spec.rb

```diff
@@ -0,0 +1,259 @@
+
+    it "adds a new reservation to array of all reservations" do
+
+      expect(@reservation_hub.reservations.length).must_equal 2
+
+    end
+  end
+
+
+  describe "add room block" do
+
+    before do
+      start_date = Date.new(2018,01,03)
+      end_date = Date.new(2018,01,06)
+      total_rooms = 3
+      @room_block = @reservation_hub.add_room_block(start_date, end_date, total_rooms)
+    end
+
+    it "returns a new room block" do
+      expect(@room_block).must_be_kind_of Hotel::RoomBlock
+    end
+
+    it "adds a new room block to an array of all room blocks" do
+      expect(@reservation_hub.room_blocks.length).must_equal 1
+    end
+
+    it "increments the block id by 1 for every new room block" do
+
+      start_date = Date.new(2018,02,04)
+    end_date = Date.new(2018,02,15)
+    total_rooms = 2
+
+    @room_block2 = @reservation_hub.add_room_block(start_date, end_date, total_rooms)
+
+    expect(@room_block2.block_id).must_equal 2
+    end
+  end
+
+
+  describe "create date array" do
+
+    before do
+      start_date = Date.new(2018,01,06)
+      @end_date = Date.new(2018,01,10)
+
+      @date_array = @reservation_hub.create_date_array(start_date, @end_date)
+
+      @test_dates = []
+      until start_date == @end_date
+        @test_dates << start_date
+        start_date +=1
+      end
+
+    end
+
+    it "returns an array of all dates" do
+      expect(@date_array).must_be_kind_of Array
+      expect(@date_array.length).must_equal @test_dates.length
+    end
+
+    it "includes all dates in the reservation, not including the end date" do
+      expect(@date_array).wont_include @end_date
+      expect(@date_array).must_equal @test_dates
+
+    end
+  end
+
+
+  describe "check available rooms method" do
+
+    it "returns an array of 20 rooms upon initialization" do
+
+      reservation_dates = [1,2]
+
+      available_rooms = @reservation_hub.check_available_rooms(reservation_dates)
+
+      expect(available_rooms.length).must_equal 20
+      expect(available_rooms).must_be_kind_of Array
+    end
+
+    it "doesn't include rooms that have already been booked" do
+
+      start_date = 1
+      end_date = 3
+
+      @reservation1 = @reservation_hub.add_reservation(start_date, end_date)
+
+      @reservation2 = @reservation_hub.add_reservation(start_date, end_date)
+
+      @reservation_dates = [1,2]
+
+
+      available_rooms = @reservation_hub.check_available_rooms(@reservation_dates)
+
+      expect(available_rooms.length).must_equal 18
+
+    end
+  end
+
+
+  describe "assign room" do
+
+    before do
+      @reservation_dates = [1,2,3]
+    end
+
+    it "will assign the first value in the array to the reservation" do
+
+      room_id = @reservation_hub.assign_room(@reservation_dates)
+      expect(room_id).must_equal 1
+
+    end
+
+    it "won't duplicate room ids if two reservations share the same dates" do
+
+      reservation_dates = [1,2,3]
+      room_id = @reservation_hub.assign_room(reservation_dates)
```

Comment:
There are a number of test cases around whether two ranges of dates overlap that you're missing here. See the slack message posted in #general.